src/scripts/main.js

```diff
@@ -1,3 +1,229 @@
+
+  if (button.classList.contains('start')) {
+    button.classList.replace('start', 'restart');
+    button.innerText = 'Restart';
+    messageStart.classList.add('hidden');
+  } else {
+    isWin = false;
+    reset();
+  }
+
+  addNumber();
+  addNumber();
+  render();
+});
+
+function checkRows() {
+  let hasDuplicates = false;
+
+  newGameField.forEach(row => {
+    if (row.some((cell, index) => cell === 0 || cell === row[index + 1])) {
+      hasDuplicates = true;
+    }
+  });
+
+  return hasDuplicates;
+}
+
+function checkColumns() {
+  let hasDuplicates = false;
+
+  newGameField.forEach(row => {
+    row.forEach((cell, index) => {
+      if (cell === row[index + 1]) {
+        hasDuplicates = true;
+      }
+    });
+  });
+
+  return hasDuplicates;
+}
+
+function reverseRows() {
+  newGameField.forEach(row => row.reverse());
+}
+
+function transposeGameField() {
+  newGameField = newGameField[0]
+    .map((x, colIndex) => newGameField.map(row => row[colIndex]));
+}
+
+function findEmptyCell() {
+  const emptyCells = [];
+
+  gameField.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      if (cell === 0) {
+        emptyCells.push([rowIndex, columnIndex]);
+      }
+    });
+  });
+
+  return emptyCells[Math.floor(Math.random() * emptyCells.length)];
+}
+
+function isPosibleToMove() {
+  if (checkRows()) {
+    return true;
+  }
+
+  transposeGameField();
+
+  return checkColumns();
+}
+
+const addNumber = () => {
+  const [randomX, randomY] = findEmptyCell();
+
+  gameField[randomX][randomY] = Math.random() < 0.9 ? 2 : 4;
+};
+
+const left = () => {
+  if (!checkRows) {
+    return;
+  }
+
+  newGameField = newGameField.map(row => {
+    const newRow = row.filter(cell => cell !== 0);
+
+    newRow.forEach((cell, index) => {
+      if (cell === newRow[index + 1]) {
+        newRow[index] *= 2;
+        newRow.splice(index + 1, 1);
+        score += newRow[index];
+
+        if (newRow[index] === 2048) {
+          isWin = true;
+        }
+      }
+    });
+
+    return newRow.concat(Array(size - newRow.length).fill(0));
+  });
+};
+
+const right = () => {
+  if (!checkRows) {
+    return;
+  }
+
+  reverseRows();
+  left();
+  reverseRows();
+};
+
+const down = () => {
+  transposeGameField();
+  right();
+  transposeGameField();
+};
+
+const up = () => {
+  transposeGameField();
+  left();
+  transposeGameField();
+};
+
+const render = () => {
+  gameField.forEach((row, rowIndex) => {
+    row.forEach((cell, colIndex) => {
+      const elem = fieldRows[rowIndex].children[colIndex];
+
+      if (cell === 0) {
+        elem.textContent = '';
+        elem.className = 'field-cell';
+      } else {
+        elem.textContent = cell;
+        elem.className = `field-cell field-cell--${cell}`;
+      }
+    });
+  });
+
+  gameScore.textContent = score;
+};
+
+const reset = () => {
+  score = 0;
+
+  gameField = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
```

Comment:
You can create const EMPTY_FIELD and use it here and at start of the page
const EMPTY_FIELD = [
    [0, 0, 0, 0],
    [0, 0, 0, 0],
    [0, 0, 0, 0],
    [0, 0, 0, 0],
```suggestion
  gameField = EMPTY_FIELD;
```
  ];